src/order/filter/at_top_bot.lean

```diff
@@ -501,22 +512,150 @@ lemma tendsto_at_bot_add_const_right (C : β) (hf : tendsto f l at_bot) :
   tendsto (λ x, f x + C) l at_bot :=
 @tendsto_at_top_add_const_right _ (order_dual β) _ _ _ C hf
 
+lemma tendsto_neg_at_top_at_bot : tendsto (has_neg.neg : β → β) at_top at_bot :=
+begin
+  simp only [tendsto_at_bot, neg_le],
+  exact λ b, eventually_ge_at_top _
+end
+
+lemma tendsto_neg_at_bot_at_top : tendsto (has_neg.neg : β → β) at_bot at_top :=
+@tendsto_neg_at_top_at_bot (order_dual β) _
+
 end ordered_group
 
-section linear_ordered_semiring
+section ordered_semiring
 
 variables [ordered_semiring α] {l : filter β} {f g : β → α}
 
 lemma tendsto_at_top_mul_at_top (hf : tendsto f l at_top) (hg : tendsto g l at_top) :
   tendsto (λ x, f x * g x) l at_top :=
 begin
   refine tendsto_at_top_mono' _ _ hg,
-  filter_upwards [hg (eventually_ge_at_top 0), hf (eventually_ge_at_top 1)],
+  filter_upwards [hg.eventually (eventually_ge_at_top 0), hf.eventually (eventually_ge_at_top 1)],
   exact λ x, le_mul_of_one_le_left
 end
 
+lemma tendsto_mul_self_at_top : tendsto (λ x : α, x * x) at_top at_top :=
+tendsto_at_top_mul_at_top tendsto_id tendsto_id
+
+/-- The function `x^n` tends to `+∞` at `+∞` for any positive natural `n`.
+A version for positive real powers exists as `tendsto_rpow_at_top`. -/
+lemma tendsto_pow_at_top {n : ℕ} (hn : 1 ≤ n) : tendsto (λ x : α, x ^ n) at_top at_top :=
+begin
+  refine tendsto_at_top_mono' _ ((eventually_ge_at_top 1).mono $ λ x hx, _) tendsto_id,
+  simpa only [pow_one] using pow_le_pow hx hn
+end
+
+end ordered_semiring
+
+section ordered_ring
+
+variables [ordered_ring α] {l : filter β} {f g : β → α}
+
+lemma tendsto_at_top_mul_at_bot (hf : tendsto f l at_top) (hg : tendsto g l at_bot) :
```

Comment:
I am surprised you did not put a name that would work with dot notation here. Something like `filter.tendsto.at_top_mul_at_bot`? Same thing for the next ones and also for `tendsto_at_top_mul_at_top` above.